src/Console/Command.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.6.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Console;
+
+use Cake\Datasource\ModelAwareTrait;
+use Cake\Log\LogTrait;
+use Cake\ORM\Locator\LocatorAwareTrait;
+use RuntimeException;
+
+/**
+ * Base class for console commands.
+ */
+class Command
+{
+    use LocatorAwareTrait;
+    use LogTrait;
+    use ModelAwareTrait;
+
+    /**
+     * The name of this command. Inflected from the class name.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * Constructor
+     *
+     * By default CakePHP will construct command objects when
+     * building the CommandCollection for your application.
+     */
+    public function __construct()
+    {
+        $locator = $this->getTableLocator() ? : 'Cake\ORM\TableRegistry';
+        $this->modelFactory('Table', [$locator, 'get']);
```

Comment:
This could be:
```php
$this->modelFactory('Table', function($alias){
    return $this->getTableLocator()->get($alias);
});
```
This way the factory will use the new locator if it was later injected using  a setter.

Reply:
Good idea.